check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+and will not run (overrides enabled).
+
+### `commands`
+`commands`: Commands exposed by this linter. [See Linter Command Definition](#linter-command-definition)
+
+### `deprecated`
+
+`deprecated`: *string*. Deprecated information (set when linter is deprecated)
+
+### `known_bad_versions`
+
+`known_bad_versions`: *string list*. Versions of a linter that are known to be broken 
+and should not be run with Trunk. We will fall back to a `known_good_version` 
+if init or upgrade chooses something in this set.
+
+### `plugin_url`
+
+`plugin_url`: *string*, a plugin url for reporting issues.
+
+### `query_compile_commands`
+
+`query_compile_commands`, *optional boolean*
+
+### `idempotent`
+
+`idempotent`: *optional boolean*. Indicates whether a linter is idempotent w.r.t. config + 
+source code inputs.  e.g. semgrep fetches rules from the internet, so it's not idempotent . If 
+set, causes cache_ttl to default to 24h
+
+### `cache_ttl`
+
+`cache_ttl`, duration string, How long cached results are kept before they expire
+
+### `target`
+
+`target`: optional string, what target does this run on
+
+### `suggest_if`
+
+How to determine if this linter should be auto-enabled/recommended. Possible 
+values are `never`, `config_present`, and `files_present`.
+
+* `config_present` will auto-enable a linter if Trunk sees any [`direct_config`](#direct_configs) for it .
+* `files_present` will auto-enable a linter if Trunk sees any file type that it operates on.
+* `never` will never auto-enable this linter.
+
+Trunk curates the values of `suggest_if` for all linters in the [plugins](https://github.com/trunk-io/plugins) repo.
+
+### `supported_platforms`
+
+Platform constraint. If incompatible, renders a notice.
+
+### `tools`
+
+`tools`, string list
+
+### `main_tool`
+
+`main_tool`, *string*. The linter's main tool. This should be the tool that provides the binary 
+for the linter. It will generally be the same as the linter's name, but may differ if the linter 
+is a wrapper. The main tool is used to determine the linter's version.
+
+### `path_format`
+
+`path_format`, Whether to use the platform-specific paths or generic "/". Default native.
+
+
+
+## Linter Command Definition
+
+The linter command definitions are defined in `lint.linter.commands`. A single linter can have 
+multiple commands if it is used in different ways.  
+
+*Note:*. If you define the executable to run here (the command definition), then you should *not*
+define it also in the linter definition. Defining it here as a command is preferred.
+
+### `name`
+
+`name`: *string*. A unique name for this command (some tools expose multiple commands, format, 
+lint, analyze, etc.).
+
+### `output`
+
+`output`: which type of output this linter produces. [See Output Types](output-types.md).
+
+### `prepare_run`
+
+`prepare_run`: an extra command to run before running a linter.
+
+### `run`
+
+`run`: The command to run a linter. 
```

Comment:
Do we have a table of all possible template expansions for `run` somewhere?

Reply:
Do you mean the variables that are exposed?  We have this list. https://docs.trunk.io/check/configuration/custom-linters#command

Reply:
I added an example of how dart uses the `${target}` variable, and also linked to the list of all variables.